internal/cli/server/config.go

```diff
@@ -66,6 +69,8 @@ type Config struct {
 	// Tracer configuration (JSON)
 	VMTraceJSONConfig string `hcl:"vmtrace.jsonconfig,optional" toml:"vmtrace,optional"`
 
+	TracerHooks *tracing.Hooks
+
```

Comment:
Unsure about the need for this, but don't think it would remain.

I think you will leverage indeed the `config.VMTrace` like you are doing, but there will no need to keep `TracerHooks` in the Config instantiated.

The `Blockchain` should have the hooks set on it and it will hold the hooks instance.